integrations/api_repo_test.go

```diff
@@ -46,6 +47,113 @@ func TestAPISearchRepoNotLogin(t *testing.T) {
 		assert.Contains(t, repo.Name, keyword)
 		assert.False(t, repo.Private)
 	}
+
+	user := models.AssertExistsAndLoadBean(t, &models.User{ID: 15}).(*models.User)
+	user2 := models.AssertExistsAndLoadBean(t, &models.User{ID: 16}).(*models.User)
+	orgUser := models.AssertExistsAndLoadBean(t, &models.User{ID: 17}).(*models.User)
+
+	type privacyType int
+
+	const (
+		_ privacyType = iota
+		privacyTypePrivate
+		privacyTypePublic
+	)
+
+	// Map of expected results, where key is user for login
+	type expectedResults map[*models.User]struct {
+		count       int
+		repoOwnerID int64
+		repoName    string
+		privacy     privacyType
+	}
+
+	testCases := []struct {
+		name, requestURL string
+		expectedResults
+	}{
+		{name: "RepositoriesMax50", requestURL: "/api/v1/repos/search?limit=50", expectedResults: expectedResults{
+			nil:   {count: 12, privacy: privacyTypePublic},
+			user:  {count: 12, privacy: privacyTypePublic},
+			user2: {count: 12, privacy: privacyTypePublic}},
+		},
+		{name: "RepositoriesMax10", requestURL: "/api/v1/repos/search?limit=10", expectedResults: expectedResults{
+			nil:   {count: 10, privacy: privacyTypePublic},
+			user:  {count: 10, privacy: privacyTypePublic},
+			user2: {count: 10, privacy: privacyTypePublic}},
+		},
+		{name: "RepositoriesDefaultMax10", requestURL: "/api/v1/repos/search", expectedResults: expectedResults{
+			nil:   {count: 10, privacy: privacyTypePublic},
+			user:  {count: 10, privacy: privacyTypePublic},
+			user2: {count: 10, privacy: privacyTypePublic}},
+		},
+		{name: "RepositoriesByName", requestURL: fmt.Sprintf("/api/v1/repos/search?q=%s", "big_test_"), expectedResults: expectedResults{
+			nil:   {count: 4, repoName: "big_test_", privacy: privacyTypePublic},
+			user:  {count: 4, repoName: "big_test_", privacy: privacyTypePublic},
+			user2: {count: 4, repoName: "big_test_", privacy: privacyTypePublic}},
+		},
+		{name: "RepositoriesAccessibleAndRelatedToUser", requestURL: fmt.Sprintf("/api/v1/repos/search?uid=%d", user.ID), expectedResults: expectedResults{
+			// FIXME: Should return 4 (all public repositories related to "another" user = owned + collaborative), now returns only public repositories directly owned by user
```

Comment:
but here is no user login, what's collaborative?

Reply:
Repositories related to `user.ID`. In my opinion logged user is here only to check rights. That means logged user and anonymous user has same rights if searching for repositories related to another user.

Reply:
But collaborative also includes repositories the user has contributed to. Don't think thats right.
IMO the question is: transitive owning!? (user owns org, org owns repo => user owns repo?) @bkcsoft and me have discussed that already in discord.

Reply:
@daviian It includes repositories, for which user has some rights through teams. In dashboard, `collaborative` is defined as `ownerID != userID` and `Source` as `ownerID == userID`. You don't know that user is owner of organization from api/repo/search result.

Reply:
If you own organization, you are automaticaly in `Owners` team.

Reply:
I think there should be a vote about the expected behaviour of this API call.
@go-gitea/owners

Reply:
@lunny @Morlinest I compared the result to githubs implementation. Github doesn't include any repositories from organizations (even owned ones) when searching for repositories of a user. So I think `2` is correct.

Please add check for `repoOwnerID`

Reply:
Since we need collaborative for our dashboard we should include it in "public" search as well, but limited to public orgs and membership and public repositories.